62.CAD/vertex_shader.hlsl

```diff
@@ -336,17 +346,79 @@ PSInput main(uint vertexID : SV_VertexID)
-        What you basically need to do here is transform the box min,max and set `outV.position` correctly based on that + vertexIdx
-        and you need to do certain outV.setXXX() functions to set the correct (transformed) data to frag shader
-
-        Another note: you may know that for transparency we draw objects twice
-        only when `writeToAlpha` is true (even provoking vertex), sdf calculations will happen and alpha will be set
-        otherwise it's just a second draw to "Resolve" and the only important thing on "Resolves" is the same `outV.position` as the previous draw (basically the same cage)
-        So if you do any precomputation, etc for sdf caluclations you could skip that :D and save yourself the trouble if `writeToAlpha` is false.
-    */
+    else if (objType == ObjectType::CURVE_BOX)
+    {
+        outV.setColor(lineStyle.color);
+        outV.setLineThickness(screenSpaceLineWidth / 2.0f);
+
+        CurveBox curveBox;
+        curveBox.aabbMin = vk::RawBufferLoad<double2>(drawObj.geometryAddress, 8u);
+        curveBox.aabbMax = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2), 8u);
+        for (uint32_t i = 0; i < 3; i ++)
+        {
+            curveBox.curveMin[i] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * (2 + i), 8u);
+            curveBox.curveMax[i] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * (2 + 3 + i), 8u);
+        }
+
+        const double2 ndcAabbExtents = double2(
+            length(transformVectorNdc(clipProjectionData.projectionToNDC, double2(curveBox.aabbMax.x, curveBox.aabbMin.y) - curveBox.aabbMin)),
+            length(transformVectorNdc(clipProjectionData.projectionToNDC, double2(curveBox.aabbMin.x, curveBox.aabbMax.y) - curveBox.aabbMin))
+        );
+        // Max corner stores the quad's UVs:
+        // (0,1)|--|(1,1)
+        //      |  |
+        // (0,0)|--|(1,0)
+        double2 maxCorner = double2(bool2(vertexIdx & 0x1u, vertexIdx >> 1));
+        
+        // Anti-alising factor + 1px due to aliasing with the bbox (conservatively rasterizing the bbox, otherwise
+        // sometimes it falls outside the pixel center and creates a hole in major axis)
+        // The AA factor is doubled, so it's dilated in both directions (left/top and right/bottom sides)
+        const double2 dilatedAabbExtents = ndcAabbExtents + 2.0 * ((globals.antiAliasingFactor + 1.0) / double2(globals.resolution));
+        // Dilate the UVs
+        maxCorner = ((((maxCorner - 0.5) * 2.0 * dilatedAabbExtents) / ndcAabbExtents) + 1.0) * 0.5;
+        const double2 coord = transformPointNdc(clipProjectionData.projectionToNDC, lerp(curveBox.aabbMin, curveBox.aabbMax, maxCorner));
+        outV.position = float4((float2) coord, 0.f, 1.f);
+
+        const uint major = (uint)SelectedMajorAxis;
+        const uint minor = 1-major;
+
+        nbl::hlsl::shapes::Quadratic<double> curveMin = nbl::hlsl::shapes::Quadratic<double>::construct(
```

Comment:
construct with <float> and unpack uint to float here before giving it to Quadrtic<float>::construct

make `pack/unpackCurveBoxValue` in common.hlsl which you can use both in hlsl and c++

Reply:
c++ one has issues: https://discord.com/channels/593902898015109131/723305695046533151/1168663165991661711